src/services/qrService.ts

```diff
@@ -205,19 +198,90 @@ const connectSiopV2 = async (args: IQrDataArgs): Promise<void> => {
       });
       showToast(ToastTypeEnum.TOAST_SUCCESS, {
         title: translate('credentials_share_success_toast_title'),
-        message: format(translate('credentials_share_success_toast_message'), verifier)
+        message: format(translate('credentials_share_success_toast_message'), translateCorrelationIdToName(url.hostname))
       });
     })
     // TODO make human-readable message
     .catch((error: Error) => showToast(ToastTypeEnum.TOAST_ERROR, { message: error.message }));
   };
 
-  RootNavigation.navigate(ScreenRoutesEnum.CREDENTIALS_REQUIRED, {
-    verifier,
-    // TODO currently only supporting 1 presentation definition
-    presentationDefinition: request.presentationDefinitions[0].definition,
-    onSend: async (credentials: Array<VerifiableCredential>) => authenticate(() => onSend(credentials))
-  });
+  const selectRequiredCredentials = async () => {
+    await setTimeout(async () => {
+      const request: VerifiedAuthorizationRequest = await siopGetRequest({...config, id: uuidv4()});
+
+      const contacts = await getContacts({
+        filter: [{
+          identities: {
+            identifier: {
+              correlationId: url.hostname,
+            }
+          }
+        }]
+      })
+      if (contacts.length === 1) {
+        const correlationId = request.payload.client_id
```

Comment:
Since the client_id can be in the OAuth2 part (so the URI), or in the OpenID (requestObject), you should use the the getMergedProperty method. This one checks the requestObject first and then uses the OAuth2 value if not present.

So something like
````typescript
request.authorizationRequest.getMergedProperty('client_id')
````

Reply:
ah nice, used that